src/app/shared/reducers/list.reducers.ts

```diff
@@ -0,0 +1,53 @@
+import {createReducer, on} from "@ngrx/store";
+import * as TaskListActions from "../actions/list.actions";
+import {TaskModel} from "../models/task.model";
+
+export const initialTasks = {
+  taskList: [] as TaskModel[],
+  openedTask: {} as TaskModel
+}
+
+export const taskReducer = createReducer(
+  initialTasks,
+  on(TaskListActions.viewAll, (state) => ({
+      ...state,
```

Comment:
I don't think these non success actions need any state updating done, as you don't actually update anything until the action is successful, right? These initial actions are meant for triggering some kind of loading mechanism that I was talking about in the earlier comments